pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/steps/DeclarativePropsStepTest.java

```diff
@@ -0,0 +1,209 @@
+                .inFolder(folder)
+                .runFromRepo(false)
+                .logContains("Docker Label is: folder_docker",
+                        "Registry URL is: https://folder.registry",
+                        "Registry Creds ID is: " + folderCred.getId()).go();
+    }
+
+    @Test
+    public void withDefaults() throws Exception {
+        Folder folder = j.createProject(Folder.class);
+        getFolderStore(folder).addCredentials(Domain.global(), folderCred);
+        getFolderStore(folder).addCredentials(Domain.global(), grandParentCred);
+        folder.addProperty(new FolderConfig("folder_docker", "https://folder.registry", folderCred.getId()));
+        expect("declarativePropsWithDefaults")
+                .inFolder(folder)
+                .runFromRepo(false)
+                .logContains("Docker Label is: other-label",
+                        "Registry URL is: https://other.registry",
+                        "Registry Creds ID is: " + grandParentCred.getId()).go();
+    }
+
+    @Test
+    public void directParentNotSystem() throws Exception {
+        GlobalConfig.get().setDockerLabel("config_docker");
+        GlobalConfig.get().setRegistry(new DockerRegistryEndpoint("https://docker.registry", globalCred.getId()));
+        Folder folder = j.createProject(Folder.class);
+        getFolderStore(folder).addCredentials(Domain.global(), folderCred);
+        folder.addProperty(new FolderConfig("folder_docker", "https://folder.registry", folderCred.getId()));
+        expect("declarativeProps")
+                .inFolder(folder)
+                .runFromRepo(false)
+                .logContains("Docker Label is: folder_docker",
+                        "Registry URL is: https://folder.registry",
+                        "Registry Creds ID is: " + folderCred.getId())
+                .logNotContains("Docker Label is: config_docker",
+                        "Registry URL is: https://docker.registry",
+                        "Registry Creds ID is: " + globalCred.getId()).go();
+    }
+
+    @Test
+    public void grandParent() throws Exception {
+        Folder grandParent = j.createProject(Folder.class);
+        getFolderStore(grandParent).addCredentials(Domain.global(), grandParentCred);
+        grandParent.addProperty(new FolderConfig("parent_docker", "https://parent.registry", grandParentCred.getId()));
+        Folder parent = grandParent.createProject(Folder.class, "testParent"); //Can be static since grandParent should be unique
+        expect("declarativeProps")
+                .inFolder(parent)
+                .runFromRepo(false)
+                .logContains("Docker Label is: parent_docker",
+                        "Registry URL is: https://parent.registry",
+                        "Registry Creds ID is: " + grandParentCred.getId()).go();
+    }
+
+    @Test
+    public void grandParentOverride() throws Exception {
+        Folder grandParent = j.createProject(Folder.class);
```

Comment:
It's testing that it's getting `parent`'s config, rather than `grandParent` - looks that way to me?